### DISCUSSION TITLE
Kan någon göra en instruktionsfilm?

### DISCUSSION THREAD

**Discussion by stefansundin** [stefansundin/privatkopiera#63] [Ideas]
Hej allihopa,

Jag fick ganska nyligen en recension på Firefox Add-ons med en stjärna på grund av att användaren hade lite svårt att förstå hur man använder tillägget. https://addons.mozilla.org/en-US/firefox/addon/privatkopiera/reviews/1653038/

Jag undrar om det kanske är dags att göra en instruktionsfilm. Det finns en lista med sex steg längst upp på hemsidan, men jag tror att det är ganska enkelt att glömma bort att gå tillbaka till den listan efter att man har lämnat sidan för att ladda ned tillägget.

Om man kunde göra en instruktionsfilm som förklarar hur man använder Privatkopiera så tror jag att vi kan hjälpa folk som har aldrig använt kommandotolken förut. Att göra film är inte mitt specialistområde, men det finns nog ett par av er som är rätt bra på det. Om jag kunde få hjälp med det här så skulle det uppskattas stort.

Jag hade tänkt mig något i följande stil.

Skärminspelning där följande händer:

  1. Tillägget installeras.
  2. FFmpeg laddas ned och installeras.
  3. Hitta en video att ladda ned (t.ex. kort nyhetsklipp, något utan strikt copyright).
  4. Öppna Privatkopiera, välj kvalité, kopiera kommandot.
  5. Öppna en kommandotolk. Zooma in videon på kommandotolken.
  6. Klistra in kommandot i kommandotolken. Förklara kort hur man hittar `ffmpeg.exe`.
  7. Vänta tills klippet har laddats ned (kanske snabba upp videon här om det tar lång tid).
  8. Öppna filen i en videospelare och spela upp det i några sekunder.

Noteringar med förklarande text kan visas under videons gång. Alternativt förklara med röstinspelning. Men det är inte garanterat att folk har igång ljud/högtalare.

Operativsystem: Windows (enligt Firefox Add-ons så använder 73% Windows). Använd inte dark mode.

Detta är vad jag tror skulle bli bäst, men om du vill försöka dig på detta så får du gärna ändra på detaljerna om du tycker det blir bättre på något annat vis.

Om klippet blir bra så tänkte jag ladda upp det till hemsidan i mp4 format, och kanske även ladda upp klippet till YouTube. Du kan även ladda upp det till YouTube själv så kan jag länka/embedda videon på hemsidan.

Hoppas att någon kan hjälpa med detta. Tack!

**Comment by xlund** [comment]
Om det här inte redan är gjort är det något jag skulle kunna kika på under våren!

**Reply by stefansundin** [reply]
Det är inte gjort. Tack för att du är villig att göra ett försök!

**Reply by rasmus-bremholm** [reply]
Jag är också sugen, skulle kunna göra en i helgen, vill du ha den publicerad till youtube eller hur sker leverans?

**Reply by stefansundin** [reply]
YouTube går bra. Du kan även testa ladda upp den direkt här. GitHub ska stöda videoklipp i kommentarerna nu tror jag, men det kanske finns begränsningar på storlek på filen. Du kan även dela på Google Drive eller liknande.

**Comment by rasmus-bremholm** [comment]
Toppen, ska spela in i helgen och så får du se vad du tycker!

**Comment by MatteusGitHub** [comment]
Vill du att jag gör en för Mac?

**Reply by stefansundin** [reply]
Jag accepterar allt och alla försök. :)

Jag kan kanske detektera vilket OS som användaren har och embedda den bästa videon, och sen ha länkar till de andra.

**Comment by SuleymanXXX** [comment]
Har köpt en ny dator, och försökte memorera hur jag 2018 installerade ffmpeg. Men minnet är kort, är även ny på Github.
Har försökt hitta instruktionsfilmen, men hittar inte den. Hur hittar jag länken till Instruktionsfilmen?

**Reply by xlund** [reply]
Jag har varit i processen att byta jobb och dator så har inte hunnit fixa någon video!

Hur du installerar ffmpeg beror på vilken dator och operativsystem du har. Stefan har skrivit en bra guide på privatkopieras hemsida:

https://stefansundin.github.io/privatkopiera/#ffmpeg

—

På windows behöver du ladda ner en statisk .exe-fil som du kan hitta här:

https://www.gyan.dev/ffmpeg/builds/ffmpeg-release-essentials.zip

och för mac kan du ladda ner en build här:

https://evermeet.cx/ffmpeg/

Notera att på MacOS kommer du få en varning om att installera programvara som inte är certifierad av Apple efter att du öppnat filen för första gången.

För att kunna installera programmet behöver du gå till **Systeminställnignar** -> Säkerhet och integritet -> Generellt. Där kommer det finnas en knapp längst ned i fönstret som säger “Öppna ändå”. Efter det kommer programmet starta som vanligt.


—

Det enklaste sättet enligt mig om du använder MacOS är att använda Homebrew som är en paketinstallerar för MacOS, men den installerar mer än bara ffmpeg. Om du har homebrew installerat kan du använda dig av brew install ffmpeg i terminalen för att installera!

Se länken nedan för mer information om Homebrew och hur du kan installera det om det är något som intresserar.

https://brew.sh/index_sv

Allt gott!

**Reply by SuleymanXXX** [reply]
stort tack för snabb återkoppling. Jag fick ta hjälp av youtube och ett ett tag så lite pysslande  fixade jag det :-)
Glömde nämna att jag har Windows 11.
Nedan är länkarna till hur man kan gå till väga, o hoppas det är ok att länka till youtube här.

Windows 11
https://www.youtube.com/watch?v=jZLqNocSQDM

Windows 10
https://www.youtube.com/watch?v=r1AtmY-RMyQ

**Reply by xlund** [reply]
Toppen att det löste sig! 

Kanske de två ovan duger för Windows i dagsläget @stefansundin? Jag kan bara spela in till MacOS!

**Reply by stefansundin** [reply]
Ja, du har nog rätt. Jag tar och länkar den till Windows 11. Tack för påminnelsen.

**Reply by jonatan-holmgren** [reply]
Behöver ni en till några olika Linux Distros? Debian? Arch? Fedora? Ser ut som om ni grejar det här med MacOS och Windows, men har möjlighet till alla tre. (Gud va jag älskar KVM-QEMU)

**Reply by stefansundin** [reply]
Jag godtar alla försök. Jag vet inte vilka distributioner som folk använder dock. Det är utan tvekan minst andel användare på Linux. Statistiken på addons.mozilla.org rapporterar ~200 användare, och chrome rapporterar ~500 användare. Och jag gissar på att de flesta vet hur man gör redan eftersom de använder Linux. :)

Men gör ett klipp om du har lust, jag tar emot. :)
